specification/botservice/resource-manager/Microsoft.BotService/preview/2023-09-15-preview/networksecurityperimeter.json

```diff
@@ -0,0 +1,309 @@
+        },
+        "provisioningIssues": {
+          "description": "List of Provisioning Issues if any",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ProvisioningIssue"
+          },
+          "x-ms-identifiers": []
+        },
+        "networkSecurityPerimeter": {
+          "$ref": "#/definitions/NetworkSecurityPerimeter"
+        },
+        "resourceAssociation": {
+          "$ref": "#/definitions/ResourceAssociation"
+        },
+        "profile": {
+          "$ref": "#/definitions/Profile"
+        }
+      }
+    },
+    "ProvisioningIssue": {
+      "description": "Describes Provisioning issue for given Network Security Perimeter configuration",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Name of the issue",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Properties of Provisioning Issue",
+          "type": "object",
+          "readOnly": true,
+          "properties": {
+            "issueType": {
+              "description": "Type of Issue",
+              "type": "string"
+            },
+            "severity": {
+              "description": "Provisioning state of Network Security Perimeter configuration propagation",
+              "enum": [
+                "Warning",
+                "Error"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "Severity",
+                "modelAsString": true
+              }
+            },
+            "description": {
+              "description": "Description of the issue",
+              "type": "string"
+            },
+            "suggestedResourceIds": {
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "readOnly": true,
+              "description": "ARM IDs of resources that can be associated to the same perimeter to remediate the issue."
+            },
+            "suggestedAccessRules": {
+              "description": "Access rules that can be added to the same profile to remediate the issue.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/NspAccessRule"
+              },
+              "x-ms-identifiers": []
+            }
+          }
+        }
+      }
+    },
+    "NetworkSecurityPerimeter": {
+      "description": "Information about Network Security Perimeter",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "id": {
+          "description": "Fully qualified identifier of the Network Security Perimeter",
+          "type": "string"
+        },
+        "perimeterGuid": {
+          "description": "Guid of the Network Security Perimeter",
+          "type": "string"
+        },
+        "location": {
+          "description": "Location of the Network Security Perimeter",
+          "type": "string"
```

Comment:
this is a bit confusing , is the nwtworksecurityparameter an arm resource ? if you have a reference to it here then why are you duplicating its properties like location here ?

Reply:
A resource's location can be different with location of network security perimeter it belongs to. This is by Azure Networking team's design.

And here, this is metadata for the network security perimeter.

For detailed schema, you can refer: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-networking/sdn-dbansal/azure-virtual-network-manager/azure-virtual-network-manager/develop/design-docs/nsp/network-security-perimeter-design-specification#effective-nsp-configuration-api

Reply:
got it thanks